pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,192 @@ def __call__(self, intg):
+
+            # Merge the metadata
+            self._merge_meta(out, merged_times)
+
+    def _active_files(self):
+        times = []
+        for file, stats, cfg in self.sfiles.values():
+            cfgsect = stats.get('tavg', 'cfg-section')
+            std = cfg.get(cfgsect, 'std-mode')
+            twindow = stats.getliteral('tavg', 'range')
+
+            for w in twindow:
+                times.append((Interval(*w), (file, stats, cfg), std))
+
+        active = times
+        for i, (r, f, s) in enumerate(times):
+            if any([(r in R and f != F) for R, F, S in active]):
+                active.pop(i)
+            elif any([r.o_intersection(R) for R, F, S in active]):
+                raise ValueError('Clash of average file time windows')
+
+        return active
+
+    def _compatibility_merge(self, active):
+        for t_rng, (file, stats, cfg), std_mode in active:
+            cfgsect = stats.get('tavg', 'cfg-section')
+            fields = stats.get('data', 'fields').split(',')
+
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+
+            for k in cfg.items(cfgsect, prefix='avg-'):
+                if cfg.get(cfgsect, k) != self.cfg.get(self.cfgsect, k):
+                    raise RuntimeError('Different average field definitions')
+
+    def _eval_fun_exprs(self, avars):
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars))
+
+        # Evaluate the function and return
+        return np.array([npeval(v, subs) for v in self.fexprs])
+
+    def _init_tavg_merge(self, file, stats, cfg):
+        self.stats, self.cfg = stats, cfg
+        self.cfgsect = cfgsect = stats.get('tavg', 'cfg-section')
+
+        self.uuid = file['mesh_uuid']
+
+        self.fields = fields = stats.get('data', 'fields').split(',')
+        self.region = cfg.get(cfgsect, 'region')
+
+        self.dkeys = [k for k in file if k.startswith('tavg_')]
+
+        # Truth table to extract collumns later
+        self.aidx = aidx = [f.startswith('avg-') for f in fields]
+        self.sidx = sidx = [f.startswith('std-') for f in fields]
+        self.anames = [f.removeprefix('avg-') for f in compress(fields, aidx)]
+        self.snames = [f.removeprefix('std-') for f in compress(fields, sidx)]
+        nsnames = len(self.snames)
+
+        # Build fun avg expressions
+        c = cfg.items_as('constants', float)
+        self.fexprs = fexprs = []
+        for k in cfg.items(cfgsect, prefix='fun-avg-'):
+            fexprs.append(cfg.getexpr(cfgsect, k, subs=c))
+
+        # Initialise std max and avg registers
+        if self.std_all:
+            self.std_max = np.zeros(nsnames)
+            self.std_avg = np.zeros(nsnames)
+
+    def _merge_data(self, out, active, avg_time):
+        a = active
```

Comment:
Make a copy as the caller probably does not expect the active list to change after the function has been called.